packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -220,25 +251,29 @@ export class DeploymentManager {
       const service = interpret(machine)
         .onTransition(async state => {
           if (state.changed) {
-            this.spinner.render();
-            if (state.matches('idle')) {
-              this.logSpinnerMessage('rollback', `Starting rollback`);
-            } else if (state.matches('preRollback')) {
-              this.logSpinnerMessage('rollback', `Waiting for previous deployment to finish`);
-            } else if (state.matches('rollback.waitForTablesToBeReady')) {
-              this.logSpinnerMessage('rollback', `Waiting for DynamoDB indices to be ready`);
-            } else if (state.matches('rollback')) {
-              this.logSpinnerMessage('rollback', `Rolling back (${maxDeployed - state.context.currentIndex} of ${maxDeployed})`);
+            const key = Object.keys(rollbackSpinnerMessage).find(elm => state.matches(elm));
+            if (key) {
+              const keyObj = rollbackSpinnerMessage[key];
+              this.logger(keyObj.machine, [{ spinner: keyObj.message }])();
+              this.spinner.start(keyObj.message);
```

Comment:
this is duplicated code. 
Could we add top level routing to this logic that pick's message map first?
Or change key convention and have just one map?